### PR TITLE
Upgrades to the grid class(es)

### PR DESCRIPTION
**Pull request by bartgol**
This PR improves a bit the AbstractGrid interface class, adding the possibility to get more info from the grid (e.g., the number of vertical levels). Additionally, it adds a SimpleGrid class, which makes testing of stand-alone parametrizations (or all physics, but no dynamics) a bit easier.

### CODE REVIEW

File: components/scream/data/scream_input.yaml [hunk 1]
```diff
@@ -16,4 +16,5 @@ Atmosphere Driver:
     Reference Grid: SE Physics
```
Comment (AaronDonahue) [inline components/scream/data/scream_input.yaml:16]: Is SE Physics deprecated now as a reference grid?  I get the following error when unit testing:
```
  /p/lustre2/donahue5/E3SM_simulations/SCREAM/scream_development/
  rain_evap_tests/scream/components/scream/src/physics/share/
  physics_only_grids_manager.cpp:33: FAIL:
  grid_name=="Physics Only" || grid_name=="Physics"
  Error! Only 'Physics Only' (and, for convenience, 'Physics') grid supported
  for physics-only runs.
         Requested grid: SE Physics
```

which goes away if I delete SE.

**Review by AaronDonahue — CHANGES_REQUESTED** [review]

**Comment by PeterCaldwell** [comment]
My understanding from Aaron is that #554 depends on this fix, so I'd appreciate it if it was prioritized...

File: components/scream/data/scream_input.yaml (hunk 1 above)
Comment (bartgol) [reply components/scream/data/scream_input.yaml:16]: The "Physics Only" grid manager does not produce a SEGrid, which is why "SE Physics" won't work. I will fix the ad input file.

**Review by AaronDonahue — APPROVED** [review]

**Comment by AaronDonahue** [comment]
@bartgol , thanks for putting all of these changes in.

**Pull request merged**